src/Auth/types.ts

```diff
@@ -0,0 +1,13 @@
+export type User = {
+    readonly id: number;
+    readonly email: string;
+    readonly name: string;
+    readonly lastname: string;
+};
+
+export type AuthService = {
+    readonly register: (email: string, password: string) => Promise<User> | unknown;
```

Comment:
I'm not sure that `unknown` is a good option here, after we send a register/login request we usually get either a user object or an error.
It might we we can use something like [`Null Object`](https://refactoring.guru/introduce-null-object) here. But we can name it `AnonymousUser`.
What do you think?

Reply:
I think that Promise<User> is enough, it was just my fault to add unknown.